Cleanup

daveisfera opened the pull request:
Apply fixes found by `pyupgrade` and `ruff`

Eittipat commented:
@daveisfera Thank you for pull request. Your pull request is cool. 😎

I think `ruff` package should be added to `requirements-dev.txt`, so we can use it without docker while developing. 
Also,`ruff format` and `ruff format --check` should be added into Makefile along with other commands 

It will be wonderful if you can integrate ruff checking into current workflows, run checking first and then testing.

Thank you

daveisfera commented:
I added `ruff` to the local workflow but left the checks as separate GitHub Actions. I personally like keeping them separate because then it's faster and easier to see what exactly is wrong. You can then make them required status checks to merge a PR, but glad to add them to the existing test workflow, if that's what you'd prefer.

I also added a matrix strategy so the tests will run on all supported versions of Python and I'd recommend switching the `on` condition to `push` so it will run on everything that's submitted automatically
